### ISSUE TITLE
How do you feel about porting these schemes to other editors?

### ISSUE THREAD

**Issue by BrianHicks** [Yummygum/city-lights-syntax-vsc#57]
City Lights is such a nice set of colors, and I'd like to use it myself. However, I don't use VSCode, Atom, or Sublime. I would like to make a theme for my own editor, but I don't think I can as you specify NoDerivatives in the license. Would you consider dropping this clause? I have seen elsewhere you want to retain control of the icons—which is totally understandable—so what about only for the color scheme itself?

**Comment by BrianHicks** [comment]
based on "Be my guest" in https://github.com/Yummygum/city-lights-syntax-vsc/issues/36, I'm going to proceed with this work for myself. A little further reading makes me think this may be an adaptation instead of a derivative since it does not require any original creativity, so it may be fine under the license terms. I'd still like clarification as to your intent, if you don't mind. 😄

**Comment by noudadrichem** [comment]
Hi @BrianHicks !

Glad you are liking the color scheme and want to support our theme! 🌃
We're very supportive on porting this to other editors but we do like to keep this in our own hands. Therefore we would like to work in forks. The guy in #36 did not supply us with a pull request but I did make a repository for Jetbrains. [You can find that repo here](https://github.com/Yummygum/city-lights-jetbrains).
If you fork this repo we would be very happy because we can also put this on our [website](http://citylights.xyz)!
Tyvm in advance and I'm looking forward to see what you made! 🧙‍♂️

tl;dr. Be my guest! But fork it from our account. If the repo isn't there yet @ me and I'll make a repo for the editor you want to port it to!

**Comment by BrianHicks** [comment]
To be honest, this kind of raises red flags for me—what do you want to keep control of? Are you asking me, as a contributor, to totally assign copyright to you for my work? If you're open to forks and derivatives like this, why have you specified CC by-nc-nd?

I hope this doesn't come across as unfriendly; I want this to exist! I just want to make sure we're not entering into some ad-hoc contributor/maintainer relationship where we have not specified what is OK and what is not.

**Comment by noudadrichem** [comment]
I'm not asking to assign complete copyrights. But Yummygum do likes to have all City Lights related stuff in one place. That's what I know (not working there anymore). 
We're not the baddest of people. So getting mentions that you made it for that specific editor is something I see as normal. 

What prevents you exactly? I also want this to exist! Jetbrains is something I would like to see aswel, and I know a lot of people who want it too!

**Comment by BrianHicks** [comment]
here's how I understand it:

by-nc-nd explicitly disallows derivative works. In a project like this, that means no PRs or ports by default, since they're exactly that: the original work plus new creative effort.

This is kind of fine a case-by-case basis if you get special permission (like Crockford allowing IBM to use jslint for evil.) I maybe have that in your message above, but the copyright statement says that Yummygum is the copyright holder. If you don't represent them anymore then I don't think I actually have a license to create a derivative work here.

All this is sort of pedantic if Yummygum is not planning to enforce the license. But I do not like to rely on good intent and mutual understanding when the legal documents reflect something else. I can take your word that you and Yummygum are wonderful humans who have nothing but goodwill towards forks, but who knows what the future holds? Companies get bought out all the time, and I have no evidence that a lax enforcement scheme will always be in effect.

**Comment by noudadrichem** [comment]
I will contact Yummygum about this. 
I'm actually not sure where the fuzz is all about. What do you want to get out of a "legal allowance" to make a port to Jet Brains editor? I just said you're able to. 😛
City Lights is free to use and we like to keep it that way. 

Either way, I'll contact Yummygum about licensing!

cc @Yummygum

**Comment by noudadrichem** [comment]
What licence would you like to see? @BrianHicks

**Comment by BrianHicks** [comment]
I mean if I could pick any license under the sun I’d recommend BSD 3-Clause. Just dropping the ND clause from this license would be fine with me though if you’re open to a change and want to stay pretty similar.

Reading some stuff above I’m not actually interested in putting to jetbrains, but Kakoune. I never actually said what target I’m interested in!

**Comment by Yummygum** [comment]
Hi @BrianHicks!

Thanks again for supporting the City Lights theme! 

I hereby grant you legal allowance to port this theme to Kakoune editor. But only if you fork from our [repo](https://github.com/Yummygum/city-lights-kakoune), as said; we like to keep City Lights related things in 1 place.

**Comment by noudadrichem** [comment]
Hi @BrianHicks,

I hope this is the reaction you wanted. I'm closing this for now.
